File: frontend/src/types/product.ts
export interface Inventory {
  id: string;
  product_id: string;
  size: string;
  quantity: number;
  reserved_quantity: number;
  location?: string;
  min_stock_level?: number;
  created_at: string;
  updated_at: string;
}

export interface Barcode {
  id: string;
  product_id: string;
  barcode_value: string;
  barcode_type: string;
  is_active: boolean;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface Product {
  id: string;
  brand_id: string;
  brand_name?: string;
  brand_icon_url?: string;
  product_code: string;
  product_name: string;
  description?: string;
  image_url?: string;
  inventory?: Inventory[];
  barcode?: Barcode;
  created_at: string;
  updated_at: string;
}

export interface ProductCreate {
  brand_id: string;
  product_code: string;
  product_name: string;
  description?: string;
}

export interface ProductUpdate {
  brand_id?: string;
  product_code?: string;
  product_name?: string;
  description?: string;
}

File: backend/app/schemas/product.py
from typing import Optional, List
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, field_validator

# Inventory 스키마
class InventorySchema(BaseModel):
    id: str
    product_id: str
    size: str
    quantity: int
    reserved_quantity: int
    location: Optional[str] = None
    min_stock_level: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    @field_validator('id', 'product_id', mode='before')
    @classmethod
    def convert_uuid_to_str(cls, v):
        if isinstance(v, UUID):
            return str(v)
        return v

    class Config:
        from_attributes = True

class ProductBase(BaseModel):
    brand_id: str
    product_code: str
    product_name: str
    category: Optional[str] = None
    description: Optional[str] = None

class ProductCreate(ProductBase):
    pass

class ProductUpdate(BaseModel):
    brand_id: Optional[str] = None
    product_code: Optional[str] = None
    product_name: Optional[str] = None
    category: Optional[str] = None
    description: Optional[str] = None

class Product(ProductBase):
    id: str
    brand_name: Optional[str] = None
    brand_icon_url: Optional[str] = None
    inventory: Optional[List[InventorySchema]] = None
    created_at: datetime
    updated_at: datetime

    @field_validator('id', 'brand_id', mode='before')
    @classmethod
    def convert_uuid_to_str(cls, v):
        if isinstance(v, UUID):
            return str(v)
        return v

    class Config:
        from_attributes = True

class ProductList(BaseModel):
    total: int
    items: list[Product]